src/Builder.php

```diff
@@ -11,277 +11,70 @@
+    public function issuedBy(string $issuer, bool $addHeader = false): Builder;
 
     /**
      * Configures the time before which the token cannot be accepted
-     *
-     * @param int $notBefore
-     * @param bool $addHeader
-     *
-     * @return Builder
      */
-    public function canOnlyBeUsedAfter(int $notBefore, bool $addHeader = false): Builder
-    {
-        return $this->setRegisteredClaim('nbf', $notBefore, $addHeader);
-    }
+    public function canOnlyBeUsedAfter(int $notBefore, bool $addHeader = false): Builder;
 
     /**
      * Configures the subject
-     *
-     * @param string $subject
-     * @param bool $addHeader
-     *
-     * @return Builder
      */
-    public function relatedTo(string $subject, bool $addHeader = false): Builder
-    {
-        return $this->setRegisteredClaim('sub', $subject, $addHeader);
-    }
-
-    /**
-     * Configures a registed claim
-     *
-     * @param string $name
-     * @param mixed $value
-     * @param bool $addHeader
-     *
-     * @return Builder
-     */
-    protected function setRegisteredClaim(string $name, $value, bool $addHeader): Builder
-    {
-        $this->with($name, $value);
-
-        if ($addHeader) {
-            $this->headers[$name] = $this->claims[$name];
-        }
-
-        return $this;
-    }
+    public function relatedTo(string $subject, bool $addHeader = false): Builder;
 
     /**
      * Configures a header item
      *
-     * @param string $name
-     * @param mixed $value
-     *
-     * @return Builder
-     *
      * @throws BadMethodCallException When data has been already signed
      */
-    public function withHeader(string $name, $value): Builder
-    {
-        if ($this->signature) {
-            throw new BadMethodCallException('You must unsign before make changes');
-        }
-
-        $this->headers[$name] = $this->claimFactory->create($name, $value);
-
-        return $this;
-    }
+    public function withHeader(string $name, $value): Builder;
 
     /**
      * Configures a claim item
      *
-     * @param string $name
-     * @param mixed $value
-     *
-     * @return Builder
-     *
      * @throws BadMethodCallException When data has been already signed
      */
-    public function with(string $name, $value): Builder
-    {
-        if ($this->signature) {
-            throw new BadMethodCallException('You must unsign before making changes');
-        }
-
-        $this->claims[$name] = $this->claimFactory->create($name, $value);
-
-        return $this;
-    }
-
-    /**
-     * Signs the data
-     *
-     * @param Signer $signer
-     * @param Key $key
-     *
-     * @return Builder
-     */
-    public function sign(Signer $signer, Key $key): Builder
-    {
-        $signer->modifyHeader($this->headers);
-
-        $this->signature = $signer->sign(
-            $this->getToken()->getPayload(),
-            $key
-        );
-
-        return $this;
-    }
+    public function with(string $name, $value): Builder;
```

Comment:
Those names were defined on #113 we're just extracting an interface. Let's define on #150 and change them on a different PR.